src/util/chains.ts

```diff
@@ -274,6 +345,84 @@ class MaticNativeCurrency extends NativeCurrency {
   }
 }
 
+function isCelo(
+  chainId: number
+): chainId is ChainId.CELO | ChainId.CELO_ALFAJORES {
+  return chainId === ChainId.CELO_ALFAJORES || chainId === ChainId.CELO;
+}
+
+class CeloNativeCurrency extends NativeCurrency {
+  equals(other: Currency): boolean {
+    return other.isNative && other.chainId === this.chainId;
+  }
+
+  get wrapped(): Token {
+    if (!isCelo(this.chainId)) throw new Error('Not celo');
+    const nativeCurrency = WRAPPED_NATIVE_CURRENCY[this.chainId];
+    if (nativeCurrency) {
+      return nativeCurrency;
+    }
+    throw new Error(`Does not support this chain ${this.chainId}`);
+  }
+
+  public constructor(chainId: number) {
+    if (!isCelo(chainId)) throw new Error('Not celo');
+    super(chainId, 18, 'CELO', 'Celo');
+  }
+}
```

Comment:
I know we discussed it yesterday but I don't see any logic to handle it. 

By default if the user passes in a NativeCurrency object as token in or token out, when we generate calldata we will add additional logic to wrap/unwrap the native currency. If a user passes in the CeloNativeCurrency object into the SOR, how do we stop router-sdk from adding logic to wrap the token?

Reply:
Actually I think it is fine as is, it doesn't do any harm and it may be useful to have both NativeCurrency object as well as Wrapped Native Currency.